src/west/commands/project.py

```diff
@@ -491,31 +491,45 @@ def _projects(args, listed_must_be_cloned=True):
         # No projects specified. Return all projects.
         return projects
 
-    # Got a list of projects on the command line. First, check that they exist
-    # in the manifest.
-
-    project_names = [project.name for project in projects]
-    nonexistent = set(args.projects) - set(project_names)
-    if nonexistent:
-        log.die('Unknown project{} {} (available projects: {})'
-                .format('s' if len(nonexistent) > 1 else '',
-                        ', '.join(nonexistent),
-                        ', '.join(project_names)))
-
-    # Return the projects in the order they were listed
+    # Listed but missing projects. Used for error reporting.
+    missing_projects = []
+
     res = []
-    for name in args.projects:
+    for project_arg in args.projects:
         for project in projects:
-            if project.name == name:
+            if project.name == project_arg:
+                # The argument is a project name
                 res.append(project)
                 break
+        else:
+            # The argument is not a project name. See if it is a project
+            # (sub)path.
+            for project in projects:
+                # The startswith() means we also detect subdirectories of
+                # project repositories. Giving a plain file in the repo will
+                # work here too, but that probably doesn't hurt.
+                if os.path.realpath(project_arg).startswith(
+                    os.path.realpath(project.abspath)):
+
+                    res.append(project)
+                    break
+            else:
+                # Neither a project name nor a project path. We will report an
+                # error below.
+                missing_projects.append(project_arg)
+
+    if missing_projects:
+        log.die('Unknown project name{0}/path{0} {1} (available projects: {2})'
+                .format('s' if len(missing_projects) > 1 else '',
+                        ', '.join(missing_projects),
+                        ', '.join(project.name for project in projects)))
 
     # Check that all listed repositories are cloned, if requested
     if listed_must_be_cloned:
         uncloned = [prj.name for prj in res if not _cloned(prj)]
         if uncloned:
             log.die('The following projects are not cloned: {}. Please clone '
-                    "them first (with 'west fetch')."
+                    "them first (with 'west clone' or 'west fetch')."
```

Comment:
Since you're here, why not just recommend west clone only? It's been discussed a lot that its results match user expectations more since there are no detached heads at first.

Reply:
I agree, I see `west fetch` as an advanced feature.

Reply:
Doesn't hurt to mention it I think. We should trust our users to be able to look up how `west fetch` works, and this is just giving them more information.

Reply:
Forcing people to make choices takes mental effort, so I don't agree with this since we are trying to get out of their way, recommend sane choices, etc. Plain west fetch in this case would lead to results that we know most users do not want.

Reply:
How about we remove the `(with 'west clone' or 'west fetch')` part and let the user figure out how to clone by looking at the doc?

Reply:
Tried to be diplomatic and added a longer explanation instead, though I think it's overkill.